### PR TITLE
sym-node: implement reshape api as Base

### PR DESCRIPTION
**Pull request by iblis17**
TODO
- [x] update `NEWs.md` after #272 merged.
- [x] update `deprecated.jl` after #272 merged.
- [x] add test cases about the special shape arg (-1, -2, ... etc)

### CODE REVIEW

File: src/symbolic-node.jl [hunk 1]
```diff
@@ -611,9 +611,108 @@ function save(filename :: AbstractString, node :: SymbolicNode)
...
+
+reshape{N}(sym::SymbolicNode, dim::NTuple{N, Integer}; kwargs...) =
+  _reshape(sym, dim, kwargs...)
+reshape(sym::SymbolicNode, dim::Integer...; kwargs...) =
+  _reshape(sym, dim, kwargs...)
+
+@inline function _reshape{N}(sym::SymbolicNode, dim::NTuple{N, Integer};
+                             reverse::Bool=false, name::String="")
+  op = _get_cached_libmx_op_handle("reshape")
+  node = _create_atomic_symbol(op.value, ["shape"], [dump_mx_param(dim)])
```
Comment (iblis17) [inline src/symbolic-node.jl]: forgot to include `reverse`

**Comment by iblis17** [comment]
now ready for review

File: src/symbolic-node.jl [hunk 2]
```diff
@@ -611,9 +611,107 @@ function save(filename :: AbstractString, node :: SymbolicNode)
...
+reshape{N}(sym::SymbolicNode, dim::NTuple{N, Integer}; kwargs...) =
+  _reshape(sym, dim; kwargs...)
+reshape(sym::SymbolicNode, dim::Integer...; kwargs...) =
+  _reshape(sym, dim; kwargs...)
+
+@inline function _reshape{N}(sym::SymbolicNode, dim::NTuple{N, Integer};
+                             reverse::Bool=false, name::String="")
+  op = _get_cached_libmx_op_handle("reshape")
+  node = _create_atomic_symbol(op.value, ["shape", "reverse"],
+                               [dump_mx_param(dim), dump_mx_param(!reverse)])
```
Comment (iblis17) [inline src/symbolic-node.jl]: `dump_mx_param` change the dim order from Julia's to C's
and we need `!reverse` to make libmxnet inteprete special shape (like `0, -1, ,,, -4`) correctly.

Comment (vchuravy) [reply src/symbolic-node.jl]: Excellent!

**Pull request merged**